### PR TITLE
Create IFP document .docx template

### PR DESCRIPTION
**Pull request by juliusfitzhugh-ccpo**
### Implement IFP generation in HOTH
- Create IFP document `.docx` template
- Create/update models for IFP and update `mediaType`
- Implement IFP document generation `ifp-document.ts`
- Update main `generate-document.ts` function

Ticket: AT-8377

### CODE REVIEW

File: document-generation/ifp-document.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        // update amounts to have commas
+        estimatedTaskOrderValue: payload.estimatedTaskOrderValue.toLocaleString("en-US"),
+        initialAmount: payload.initialAmount.toLocaleString("en-US"),
+        remainingAmount: payload.remainingAmount.toLocaleString("en-US"),
+        // construct Funding Document information to be used in template
+        fundingDocInfo: getFundingDocInfo(payload.fundingDocument),
+      },
+      cmdDelimiter: ["{", "}"],
+    })
+  );
```
Comment (zachclark-ccpo) [inline document-generation/ifp-document.ts:25]: Should there be a `try... catch` in case `createReport` results in an error?

Comment (juliusfitzhugh-ccpo) [reply document-generation/ifp-document.ts:25]: yeah, good point. I know the `errorHandlerMiddleware` and `middy` end up capturing most of the errors, so I sometimes leave out the `try/catch` blocks.

Comment (zachclark-ccpo) [reply document-generation/ifp-document.ts:25]: True, middy will probably catch it and return a 500 back to SNOW. No need to add try/catch in that case.

**Review by jasonburkert-ccpo — APPROVED** [review]
Agree with @zachclark-ccpo, clean!

**Pull request merged**